internal/handlers/pg/msg_find.go

```diff
@@ -73,6 +73,18 @@ func (h *Handler) MsgFind(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, er
 		return nil, err
 	}
 
+	maxTimeMS, err := common.GetOptionalPositiveNumber(document, "maxTimeMS")
+	if err != nil {
+		return nil, err
+	}
+
+	if maxTimeMS != 0 {
+		ctxWithTimeout, cancel := context.WithTimeout(ctx, time.Duration(maxTimeMS)*time.Millisecond)
```

Comment:
Yeah, this is the exact thing I expect this param to do. Why don't we have the same in `msg_findandmodify.go`?

Reply:
Now it's fixed! 😃